agents/fts_stage/fts_agent.py

```diff
@@ -0,0 +1,304 @@
+import socket
+import os
+import time
+import txaio
+import argparse
+
+ON_RTD = os.environ.get('READTHEDOCS') == 'True'
+if not ON_RTD:
+    from ocs import ocs_agent, site_config
+    from ocs.ocs_twisted import TimeoutLock, Pacemaker
+
+class FTSStage:
+    """
+    Class for connecting to the FTS mirror controller
+
+    Args:
+        ip_address: IP address where controller is running
+        port: Port the controller is listening on
+        timeout: communication timeout
+        speed: speed in mm/s, defaults to 25 mm/s if None
+    """
+    TRANSLATE = 1, 74.87
+    LIMS = (-74.8,74.8)
+    SPEED = 25 #mm/s
+    INFLATION = 1.10
+    SETTLE_T = 0.2
+    MIN_WAIT = 0.1
+
+    def __init__(self, ip_address, port, timeout=10, speed=None):
+        self.ip_address = ip_address
+        self.port = int(port)
+
+        self.comm = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.comm.connect((self.ip_address, self.port))
+        self.comm.settimeout(timeout)
+
+        self.send('ENABLE X\n') #Send enable command
+        data = self.comm.recv(1024) #Collect and print response
+ 
+        if data == b'%\n':
+            self.initialized = True
+        else:
+            self.initialized = False
+        
+        self.pos = None
+        if speed is not None:
+            self.SPEED = speed
+        self.speed_code = 'F%i' % self.SPEED
+        
+ 
+    def send(self, msg):
+        self.comm.sendall( bytes(msg, 'utf-8'))
+    
+    def read(self):
+        # Controller blocks reply until motion is complete; so if
+        # you're putting a timeout in here make sure it's smart/long
+        # enough.
+        return self.comm.recv(1024)
+    
+    def home(self):
+        self.send('HOME X\n')
+        time.sleep(0.1)
+        return self.read()
+    
+    def get_position(self):
+        self.send('CMDPOS X\n')
+        time.sleep(0.1)
+        out = self.read()
+        try:
+            M, B = self.TRANSLATE
+            self.pos = (float(out[1:])-B)/M
+            return True, self.pos
+        except:
```

Comment:
What's this bare `except` catching?

Reply:
Honestly no idea, that part of the code came from ACT's "agent." Looking at it I'm guessing it's protecting from getting a non-sense response from the controller.